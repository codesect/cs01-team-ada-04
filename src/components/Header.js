import React from 'react';
import styled from 'styled-components/macro';

import { Wrapper } from './GlobalStyles';
import LogoImg from '../logo.svg';

const StyledHeader = styled.header`
  background-color: ${({ theme }) => theme.headerBackground};
  box-shadow: ${({ theme }) => theme.boxShadowHeader};
`;

const Logo = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;

  @media (min-width: ${({ theme }) => theme.breakpointS}) {
    justify-content: flex-start;
  }
`;

const LogoText = styled.div`
  margin-left: 1rem;
`;

const TagLine = styled.div`
  line-height: 1.25;
`;

const Title = styled.div`
  color: ${({ theme }) => theme.logoText};
  font-size: 2rem;
  letter-spacing: -0.125rem;
  line-height: 1;
`;

function Header() {
  return (
    <StyledHeader>
      <Wrapper>
        <Logo>
          <img src={LogoImg} alt="" />
          <LogoText>
            <Title>
              <strong>Strong</strong>Pass
            </Title>
            <TagLine>Password Generator</TagLine>
          </LogoText>
        </Logo>
      </Wrapper>
    </StyledHeader>
  );
}

export default Header;
