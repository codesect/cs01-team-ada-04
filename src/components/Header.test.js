import React from 'react';
import { render } from 'enzyme';
import 'jest-styled-components';

import Header from './Header';

describe('<Header />', () => {
  it('matches snapshot', () => {
    const HeaderWithTheme = withTheme(Header);
    const wrapper = render(<HeaderWithTheme />);

    expect(wrapper).toMatchSnapshot();
  });
});
