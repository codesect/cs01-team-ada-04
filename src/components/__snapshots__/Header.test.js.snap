// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 832px;
  padding: 0.5rem 1rem;
}

.c0 {
  background-color: #fff;
  box-shadow: 0 0 10px rgba(0,0,0,0.07);
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  margin-left: 1rem;
}

.c5 {
  line-height: 1.25;
}

.c4 {
  color: #8555c5;
  font-size: 2rem;
  -webkit-letter-spacing: -0.125rem;
  -moz-letter-spacing: -0.125rem;
  -ms-letter-spacing: -0.125rem;
  letter-spacing: -0.125rem;
  line-height: 1;
}

@media (min-width:600px) {
  .c2 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt=""
        src="logo.svg"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <strong>
            Strong
          </strong>
          Pass
        </div>
        <div
          class="c5"
        >
          Password Generator
        </div>
      </div>
    </div>
  </div>
</header>
`;
